src/lunchselector/model.clj

```diff
@@ -44,15 +44,16 @@
 
 (defn online-restaurants
   "Gets the list of online restaurants from Zomato"
-  [keyword]
-  (let [online-restaurants (fetch-restaurants keyword)
-        response-body (utils/parse-response-body-map online-restaurants)]
-    (loop [restaurant-list (:restaurants response-body)
-           resta ()]
-      (if (empty? restaurant-list)
-        resta
-        (recur (rest restaurant-list)
-               (conj resta (restaurant-name (first restaurant-list))))))))
+  ([] (online-restaurants ""))
+  ([keyword]
+   (let [online-restaurants (fetch-restaurants keyword)
+         response-body (utils/parse-response-body-map online-restaurants)]
+     (loop [restaurant-list (:restaurants response-body)
+            resta ()]
+       (if (empty? restaurant-list)
+         resta
+         (recur (rest restaurant-list)
+                (conj resta (restaurant-name (first restaurant-list)))))))))
```

Comment:
Did you just want to do `(map restaurant-name (:restaurants response-body))`? Generally, prefer in-built functions in core to basic constructs like `loop/recur` or `reduce`.